pkg/vaulturl/localvault/client.go

```diff
@@ -0,0 +1,71 @@
+package localvault
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/jenkins-x/jx/pkg/helm"
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/jenkins-x/jx/pkg/vaulturl"
+	"github.com/pkg/errors"
+)
+
+// FileSystemClient a local file system based client loading/saving content from the given URL
+type FileSystemClient struct {
+	Dir string
+}
+
+// NewFileSystemClient create a new local file system based client loading content from the given URL
+func NewFileSystemClient(dir string) vaulturl.Client {
+	return &FileSystemClient{
+		Dir: dir,
+	}
+}
+
+// Read reads a named secret from the vault
+func (c *FileSystemClient) Read(secretName string) (map[string]interface{}, error) {
+	name := c.fileName(secretName)
+	exists, err := util.FileExists(name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to check if file exists %s", name)
+	}
+	if !exists {
+		return nil, fmt.Errorf("local vault file does not exist: %s", name)
+	}
+	return helm.LoadValuesFile(name)
+}
+
+// ReadObject reads a generic named object from vault.
+// The secret _must_ be serializable to JSON.
+func (c *FileSystemClient) ReadObject(secretName string, secret interface{}) error {
+	m, err := c.Read(secretName)
+	if err != nil {
+		return errors.Wrapf(err, "reading the secret %q from vault", secretName)
+	}
+	err = util.ToStructFromMapStringInterface(m, &secret)
+	if err != nil {
+		return errors.Wrapf(err, "deserializing the secret %q from vault", secretName)
+	}
+	return nil
+}
+
+// WriteObject writes a generic named object to the vault.
+// The secret _must_ be serializable to JSON.
+func (c *FileSystemClient) WriteObject(secretName string, secret interface{}) (map[string]interface{}, error) {
+	path := c.fileName(secretName)
+	dir, _ := filepath.Split(path)
+	err := os.MkdirAll(dir, util.DefaultWritePermissions)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to ensure that parent directory exists %s", dir)
+	}
+	err = helm.SaveFile(path, secret)
+	if err != nil {
+		return nil, err
```

Comment:
Maybe wrap this error

Reply:
I did consider - but the current message is pretty good https://github.com/jenkins-x/jx/blob/ccba49d49156e5c2b1d9594aa8fc339590132000/pkg/helm/helm_helpers.go#L318